core/API/DataTableManipulator/Flattener.php

```diff
@@ -137,7 +156,18 @@ private function flattenRow(Row $row, $rowId, DataTable $dataTable,
                 $dataTable->addRow($row);
             }
             $prefix = $label . $this->recursiveLabelSeparator;
-            $this->flattenDataTableInto($subTable, $dataTable, $prefix, $logo);
+
+            $report           = ReportsProvider::factory($this->apiModule, $this->getApiMethodForSubtable($this->request));
+            $subDimension     = $report->getDimension();
+            $subDimensionName = $subDimension ? str_replace('.', '_', $subDimension->getId()) : 'label' . (substr_count($prefix, $this->recursiveLabelSeparator) + 1);
```

Comment:
Is it possible to have a test for when a report has no dimension or subdimension? To trigger the labelN logic.